frontend/src/TestEdit.js

```diff
@@ -0,0 +1,267 @@
+import React, { useEffect} from 'react';
+import  { useState } from 'react';
+import {Link, useParams} from 'react-router-dom';
+import {Button,  Container, Form, FormGroup, Input, Label} from 'reactstrap';
+import AppNavbar from './AppNavbar';
+import {useQuery} from "react-query";
+
+
+
+const emptyItem = {
+    test:{
+        name: '',
+        group:null,
+        author:null,
+        duration: '',
+        id: null
+    },
+
+    questions: [{
+        question:"",
+        id:1,
+        answerVariants:[""],
+        answers:[]
+
+    } ],
+
+    added:new Set(),
+    deleted:new Set(),
+    groupName:''
+
+};
+
+function TestEdit(){
+
+    function correctItem(data){
+        data.questions.forEach(function (q) {
+            if (q.id > maxId) {
+                maxId = q.id;
+            }
+        });
+
+        data.added = new Set();
+        data.deleted = new Set();
+        data.groupName = data.test.group.name;
+        setItem(data);
+
+    }
+
+
+    const [item, setItem] = useState(emptyItem);
+    var maxId=1;
```

Comment:
Never ever use `var` :) (refer to https://dev.to/mindninjax/stop-using-var-for-declaring-variables-2p3a or similar)